File: SourceAFIS/Matching/Minutia/EdgeConstructor.cs
using System;
using System.Collections.Generic;
using System.Text;
using SourceAFIS.General;
using SourceAFIS.Templates;

namespace SourceAFIS.Matching.Minutia
{
    public sealed class EdgeConstructor
    {
        public static EdgeShape Construct(FingerprintTemplate template, int reference, int neighbor)
        {
            PolarPoint polar = Angle.ToPolar(Calc.Difference(template.Minutiae[neighbor].Position, template.Minutiae[reference].Position));
            EdgeShape edge;
            edge.Length = (short)polar.Distance;
            edge.ReferenceAngle = Angle.Difference(template.Minutiae[reference].Direction, polar.Angle);
            edge.NeighborAngle = Angle.Difference(template.Minutiae[neighbor].Direction, Angle.Opposite(polar.Angle));
            return edge;
        }
    }
}

File: java/sourceafis/src/sourceafis/matching/minutia/EdgeConstructor.java
package sourceafis.matching.minutia;

import sourceafis.general.Angle;
import sourceafis.general.Calc;
import sourceafis.general.PolarPoint;
import sourceafis.templates.Template;
/*
 *  Call to Construct can be static and no need to create EdgeConstructor object
 */
public final class EdgeConstructor
{
	   public EdgeShape Construct(Template template, int reference, int neighbor)
       {
           PolarPoint polar = Angle.ToPolar(Calc.Difference(template.minutiae[neighbor].Position, template.minutiae[reference].Position));
           EdgeShape edge=new EdgeShape();
           edge.length = (short)polar.Distance;
           edge.referenceAngle = Angle.Difference(template.minutiae[reference].Direction, polar.Angle);
           edge.neighborAngle = Angle.Difference(template.minutiae[neighbor].Direction, Angle.Opposite(polar.Angle));
           return edge;
       }
}